zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -450,6 +463,24 @@ protected InterpreterResult jobRun() throws Throwable {
         p.settings.setParams(settings.getParams());
       }
 
+      // After the paragraph is executed,
+      // need to apply the paragraph to the configuration in the
+      // `interpreter-setting.json` config
+      if (this.configSettingNeedUpdate) {
```

Comment:
The default configuration is used when the `intpText` changes.
The configuration modified by the user on the paragraph is not overwritten by the interpreter default configuration.